pipeline/types/engine.py

```diff
@@ -48,3 +48,10 @@ class MatchingOutput:
     release: ReleaseTag
     users: List[User]
     matches: List[Match]
+
+
+@dataclass
+class MatchingInternalAnalytics:  # More metrics will be added in the future
+    n_proposed_matches_per_user: List[
+        Tuple[User, int]
+    ]  # The list will be a tuple containing the user and their number of proposed matches
```

Comment:
What do you think about a dictionary instead of a list of tuples? What are the pros and cons?

Reply:
Now that you mention it, I think a dictionary would work best in this instance. At the time I was more concerned with how I should integrate the user and the associate metric together; and I thought a tuple would be good. However, a dictionary where the user is the key (since all users should be unique anyway) and the desired metric is the value would be more sensible.

Reply:
Awesome, that sounds good!